rfc/rfc-0000-move-off-monorepo.md

```diff
@@ -0,0 +1,57 @@
+ 
+As expressed by maintainers of the https://github.com/vmware-tanzu/cartographer repository it will become hard to manage it if we use it as a monorepo and continue to add more capabilities like conventions etc.
+It is expected that this RFC describes how we split the current repository so it allows the Cartographer OSS project to evolve and scale beyond the concrete capabilities the core aims to provide.
+For example splitting out from the current repo:
+
+-  the __`Runnable`__ feature so it can release independently and install standalone in a target deployment cluster,  avoids deploying all of Cartographer which contains build cluster CRDs not desired in runtime clusters
+- the __examples directory__ so we can build up a wide range of examples that include integrations with wider platform services like conventions.  This also helps encourage contribution of community supply chains for different integrations where we can add automated testing that runs against different setups.
+- the __website__ so we can add documentation for these new integrations and platform capabilities outside of the core repository.  I.e. Documenting services like the conventions or a Jenkins integration should not require a change to the core cartographer repository, this should live outside in its own repository.
+
+## Possible Solutions
+
+There seems to be only two options possible solutions:
+- __monorepo__ we could revisit the [convention proposal](https://github.com/vmware-tanzu/cartographer/pull/514) decision and look at ways to address the monorepo concerns however there seemed a general consensus on the [office hours call](https://www.youtube.com/watch?v=iCURQqV52Uw&t=1173s) that this was not desired.
+- __microservice__ As conventions are now being worked on in a different repository, we should look at splitting out parts that currently reside in https://github.com/vmware-tanzu/cartographer.
+
+Suggestions:
+
+-  __`Runnable`__ feature so it can install standalone in a target deployment cluster and avoids deploying all of Cartographer
+-  __examples directory__ so we can build up a wide range of examples that include integrations with wider platform services like conventions.
+-  __website__ so we can add documentation for these new integrations and platform capabilities outside of the core repository.  I.e. documenting a service like the convention components or a Jenkins integration should not require a change to the core cartographer repository, this should live outside in it's own repository
+- __others?__
+
+## Cross References and Prior Art
+
+Carvel: https://github.com/vmware-tanzu - Corporate sponsored organisation
+Kubernetes: https://github.com/kubernetes - Cloud Native Computing Foundation sponsored organisation
```

Comment:
Kubernetes is a curious case, as it is developed as a mono repo and released as component repos.

Reply:
yeah I guess I was trying to show that there's a wider kubernetes ecosystem that helped evolve the project.  I.e specific areas developed special interest groups that iterate at their own cadence.  That's quite a good model to help encourage wider efforts and involvement from others.

Reply:
A further discussion (probably for a separate RFC) would deal with how splitting into separate repos could trigger changes in Governance, introducing the idea of Working Groups/Feature areas (see [Harbor](https://github.com/goharbor/community/blob/main/MAINTAINERS.md ) as an example). With that approach we could define focus areas for maintainers and potentially onboard contributors for 'microteams' with a clear scope.